src/test/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacadeTest.java

```diff
@@ -563,8 +601,390 @@ void shouldNotEnrichFromPlacementWhenSiteExistsWithNoSiteNameNorSiteLocationNorS
   }
 
   @Test
-  void shouldEnrichFromPlacementWhenPlacementSpecialtyNotExists() {
+  void shouldStillEnrichPlacementWhenNoOtherSites() throws JsonProcessingException {
+    Placement placement = new Placement();
+    placement.setTisId(PLACEMENT_1_ID);
+
+    when(placementSiteService.findOtherSitesByPlacementId(
+        Long.parseLong(PLACEMENT_1_ID))).thenReturn(Set.of());
+
+    enricher.enrich(placement);
+
+    verify(placementService, never()).request(anyString());
+    verifyNoInteractions(siteService);
+
+    verify(tcsSyncService).syncRecord(placement);
+    verifyNoMoreInteractions(tcsSyncService);
+
+    Map<String, String> placementData = placement.getData();
+    Set<Map<String, String>> otherSites = objectMapper.readValue(
+        placementData.get(PLACEMENT_DATA_OTHER_SITE_NAME), new TypeReference<>() {
+        });
+    assertThat("Unexpected other sites count.", otherSites.size(), is(0));
+  }
+
+  @Test
+  void shouldStillEnrichPlacementWhenOtherSiteMissingSiteId() throws JsonProcessingException {
```

Comment:
I can see this correctly tests a conditional in the code, but I am curious whether this would ever happen: is it mainly to avoid 'blocking' a placement if it's related (other) site record is somehow defective? There are no PlacementSites in metabase without a SiteId. I guess I'm wondering if this is a special case we've included for some reason - I'm not sure if we are as defensive with other data types.

Reply:
I added a filter on null values before calling `toString()` so I wanted to make sure that branch was tested, but I don't think it's ever possible for the data to be missing in TIS.

We've been caught out by weird data quality assumptions before, like "no site/grade will ever have `id = 0`" or you can't have a `CurriculumMembership` without a matching `ProgrammeMembership`.

Reply:
True dat. Well, at least every SiteId in PlacementSite refers to a real Site - I just checked 😁